src/Mod/PartDesign/Gui/Utils.cpp

```diff
@@ -153,6 +141,85 @@ PartDesign::Body *getBody(bool messageIfNot, bool autoActivate, bool assertModer
     return activeBody;
 }
 
+PartDesign::Body * makeBodyActive(App::DocumentObject* body, App::Document *doc)
+{
+    App::DocumentObject *parent = 0;
+    std::string sub;
+
+    for(auto &v : body->getParents()) {
+        if(v.first->getDocument()!=doc)
+            continue;
+        if(parent) {
+            body = 0;
+            break;
+        }
+        parent = v.first;
+        sub = v.second;
+    }
+
+    if(body) {
+        auto _doc = parent?parent->getDocument():body->getDocument();
+        _FCMD_DOC_CMD(Gui, _doc, "ActiveView.setActiveObject('" << PDBODYKEY
+                      << "'," << Gui::Command::getObjectCmd(parent?parent:body)
+                      << ",'" << sub << "')");
+        // return Gui::Application::Instance->activeView()->
+        //     getActiveObject<PartDesign::Body*>(PDBODYKEY,topParent,subname);
+    }
+
+    return dynamic_cast<PartDesign::Body*>(body);
+}
+
+PartDesign::Body * needActiveBodyMessage (App::Document *doc,
```

Comment:
I'm not a big fan of putting this code in the generic "Utils.cpp" file, rather than following the more idiomatic Qt path of creating a standalone class "DlgActiveBody" that does all of the management of the dialog. Is there a reason to put it in Utils?

Reply:
It just happened to grow here, plus I mimicked `DlgReference` or similar. I'm not certain if it would change much, but I can try moving it to it's own class.

Reply:
I completely understand -- stuff tends to sort of "accrete" in any file called Utils 😄 . Before you know it there's 10k lines of basically unrelated (and un-reusable) code. I'm not dead-set on moving it if there's a good reason to leave it.

Reply:
I'm actually thinking of possibly moving this message to the task view instead. That way, it would be less intrusive, and we can highlight the bodies as they are selected. I should probably discuss that in the forum at some point, I guess.